packages/frontend-2/components/viewer/saved-views/panel/View.vue

```diff
@@ -252,17 +261,54 @@ const menuItems = computed((): LayoutMenuItem<MenuItems>[][] => [
 ])
 
 const wrapperClasses = computed(() => {
-  const classParts = ['flex gap-2 p-1.5 w-full group rounded-md cursor-pointer']
+  const classParts = [
+    'flex gap-2 p-1.5 w-full group rounded-md cursor-pointer relative transition-all'
+  ]
 
   if (isActive.value) {
     classParts.push('bg-highlight-2 hover:bg-highlight-3')
   } else {
     classParts.push('hover:bg-highlight-1')
   }
 
+  // Add dragging styles
+  if (isDragging.value) {
+    classParts.push('opacity-50 scale-95')
+  }
+
   return classParts.join(' ')
 })
 
+const onDragStart = (event: DragEvent) => {
+  if (!canUpdate.value?.authorized || isLoading.value) {
+    event.preventDefault()
+    return
+  }
+
+  isDragging.value = true
+
+  const dragData = {
```

Comment:
this should be TS typed so u can ensure correct props are set here and then type the dragData on the other end when receiving it, to have a type for the result